src/ssa_ir/insn.rs

```diff
@@ -0,0 +1,189 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::codegen;
+use crate::sema::ast::{CallTy, ExternalCallAccounts};
+use crate::ssa_ir::expr::{Expr, Operand};
+use crate::ssa_ir::ssa_type::InternalCallTy;
+use solang_parser::pt::Loc;
+
+use super::ssa_type::PhiInput;
+
+/// Statements using three-address code format
+#[derive(Debug)]
+pub enum Insn {
+    Nop,
+
+    /*************************** Contract As Callee ***************************/
+    // Return data to the outside callers
+    ReturnData {
+        data: Operand,
+        data_len: Operand,
+    },
```

Comment:
Many instructions are missing location information (like this one),

Always make the loc the first field, so the layout is always the same in memory

Reply:
Hi Sean, thank you for taking the time to review my PR. I noticed that certain location information is missing from some parts of the code. I believe this could be because they don't have a defined location in the `src/codegen/cfg.rs`. Is it possible that some of these parts are generated by the compiler?

Reply:
If they're missing from `codegen/cfg.rs` then that's an oversight. You can still add a `pt::Loc` to your SSA form and populate it with a bogus value with a comment saying this is missing from the CFG from.